projects/tutorial_102/src/main.c

```diff
@@ -0,0 +1,44 @@
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+typedef struct Counters {
+        uint8_t counter_a;
+        uint8_t counter_b;
+    } Counters;
+
+static void prv_timer_callback(SoftTimerId timer_id, void *context) {
+    Counters *storage = context;
+    storage->counter_a++;
+
+    LOG_DEBUG("Counter A: %d\n", storage->counter_a);
+    
+    if (storage->counter_a == (storage->counter_b)*2 + 2) {
```

Comment:
This is certainly the most innovative way to implement the counter pattern I've seen! The usual solution is just `if (storage->counter_a % 2 == 0)`. Aside, as a nitpick, the brackets around `storage->counter_b` are unnecessary.